worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,44 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function haveFailuresChanged(document) {
+    if (document.getFailures().isChanged()) {
+        
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+        
+        document.getFailures().reset();
+        
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+        
+        var newListOfFailures = new java.util.ArrayList();
+        for each (var f in listOfFailures){
+            if(!listOfOriginalFailures.contains(f)){
+                var message = {
+                    id: f.getFailureId(),
+                    stack: f.getFailureStack(),
+                    description: {
+                        source:  document.getTask().getService(com.hpe.caf.api.worker.WorkerTaskData.class).getSourceInfo().getName(),
+                        version: document.getTask().getService(com.hpe.caf.api.worker.WorkerTaskData.class).getSourceInfo().getVersion(),
+                        workflowName: document.getField("CAF_WORKFLOW_NAME"),
+                        originalDescription: f.getFailureMessage(),
```

Comment:
Could you add the action to this?

Reply:
I thought the action would have been under `source`. So, I have added it with line 218 (I may be completely wrong...)

Reply:
The action name is different from the worker name, so it could be useful to have it. I think at one stage we had the same worker in the workflow multiple times.